### PR TITLE
Adding comment for vcpkg triplet in the unique key for caching

### PR DESCRIPTION
**Pull request by sappyh**
Commenting the details about the vcpkg triplet to resolve issue #1534

### CODE REVIEW

**Comment by RickWinter** [comment]
@danieljurek Are you working on switching to vcpkg's cache?

File: eng/pipelines/templates/steps/vcpkg.yml
Comment (danieljurek) [inline eng/pipelines/templates/steps/vcpkg.yml:28]: Please remove the `#TODO` comment here. I've filed this issue to track investigating this: https://github.com/Azure/azure-sdk-for-cpp/issues/1765

File: eng/pipelines/templates/steps/vcpkg.yml
Comment (danieljurek) [inline eng/pipelines/templates/steps/vcpkg.yml:27]: This is true. However, a significant example case we're avoiding is using an [x86 build of curl](https://github.com/Azure/azure-sdk-for-cpp/blob/master/eng/pipelines/templates/jobs/archetype-sdk-client.yml#L95) on an [x64 build ](https://github.com/Azure/azure-sdk-for-cpp/blob/master/eng/pipelines/templates/jobs/archetype-sdk-client.yml#L104) of the SDK libraries. The triplet here lets us capture the target platform architecture, OS (though this is better captured in `$(Agent.Os)`), and linking configuration (static/dynamic) in one variable.  

Please update the comment with the example. This will make the value of `VCPKG_DEFAULT_TRIPLET` immediately more clear to developers who end up here later.

**Review by danieljurek — CHANGES_REQUESTED** [review]
@sappyh -- Thanks for your contribution! I've added a couple points of feedback to address before we can sign off and merge your PR. Let me know if you have any questions.

File: eng/pipelines/templates/steps/vcpkg.yml
Comment (ahsonkhan) [reply eng/pipelines/templates/steps/vcpkg.yml:27]: > Please update the comment with the example.

@danieljurek can you please give a concrete GitHub suggestion with the example you prefer to be added here.

File: eng/pipelines/templates/steps/vcpkg.yml
Comment (ahsonkhan) [reply eng/pipelines/templates/steps/vcpkg.yml:28]:
```suggestion
```

**Review by ahsonkhan — APPROVED** [review]
Looks good to me, but since we are in a code freeze while releasing our next version, please hold off on merging for the next day or so.

Thanks for the fix @sappyh

**Review by ahsonkhan — APPROVED** [review]

**Comment by RickWinter** [comment]
@ahsonkhan Are we waiting on the contributor for this?

**Comment by ahsonkhan** [comment]
Nope, just waiting for @danieljurek to approve or dismiss the `changes requested` and we can merge it :)

**Comment by ahsonkhan** [comment]
@msftbot wait until @danieljurek approves

**Review by danieljurek — APPROVED** [review]

**Pull request merged**
